crates/next-api/src/pages.rs

```diff
@@ -865,15 +868,18 @@ impl PageEndpoint {
 
                 let availability_info = Value::new(AvailabilityInfo::Root);
                 let dynamic_import_modules = collect_next_dynamic_imports(
-                    [Vc::upcast(ssr_module)],
+                    vec![Vc::upcast(ssr_module)],
                     this.pages_project.client_module_context(),
+                    VisitedDynamicImportModules::empty(),
                 )
-                .await?;
+                .await?
+                .client_dynamic_imports
+                .clone();
```

Comment:
That looks like exactly the same as above. Could we move that out of the conditional to deduplicate that?